src/screens/Users/VerifyOTP.js

```diff
@@ -0,0 +1,193 @@
+import BoxTitle from 'components/ModalBox/BoxTitle';
+import BoxBody from 'components/ModalBox/BoxBody';
+import StatusLabel from 'components/Verification/StatusLabel';
+import ResendMessage from 'components/Verification/ResendMessage';
+import CheckIcon from 'components/Verification/CheckIcon';
+
+type Props = {
+  createOneTimePassword: (walletId: string, field: Object) => void,
+  resetOneTimePassword: () => void,
+  verifyEmail: (walletId: string, code: string) => void,
+  verifyPhone: (walletId: string, code: string) => void,
+  navigation: NavigationScreenProp<*>,
+  user: User,
+  sendingOneTimePassword: boolean,
+};
+
+type State = {
+  code: string,
+};
+
+class UserSettings extends React.PureComponent<Props, State> {
+  verifyField: string;
+
+  state = {
+    code: '',
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {
+      state: { params: { field } },
+    } = props.navigation;
+
+    this.verifyField = field;
+  }
+
+  componentDidMount() {
+    this.sendOTP();
+  }
+
+  sendOTP = () => {
+    const {
+      createOneTimePassword,
+      user: { walletId },
+    } = this.props;
+
+    createOneTimePassword(walletId, {
+      smsNotification: this.verifyField === 'phone',
+    });
+  }
+
+  resetOTP = () => {
+    const { resetOneTimePassword } = this.props;
+
+    resetOneTimePassword();
+  };
+
+  confirmOTP = (code) => {
+    const {
+      verifyPhone,
+      verifyEmail,
+      user: { walletId },
+    } = this.props;
+
+    switch (this.verifyField) {
+      case 'email':
+        verifyEmail(walletId, code);
+        break;
+
+      case 'phone':
+        verifyPhone(walletId, code);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  updateCode = (code: string) => {
+    this.setState({ code }, () => {
+      if (code.length === 5) {
+        this.confirmOTP(code);
+      }
+    });
+  };
+
+  goBack = () => {
+    const { navigation } = this.props;
+
+    navigation.goBack(null);
+  };
+
+  render() {
+    const { code } = this.state;
+    const {
+      sendingOneTimePassword,
+      user,
+      user: {
+        isEmailVerified,
+        isPhoneVerified,
+      },
+    } = this.props;
+    const { verifyField } = this;
+
+    const titleText = `Enter verification code (${
+      verifyField === 'email' ? 'Email' : 'SMS'
+    })`;
+
+    const isVerified = verifyField === 'email' ?
+      isEmailVerified : isPhoneVerified;
+    const enteringCode = !isVerified && !sendingOneTimePassword;
+
+    return (
+      <ModalBox>
+        <BoxTitle title={titleText} onPressClose={this.goBack} />
+        <StatusLabel
+          user={user}
+          field={verifyField}
+          sendingOneTimePassword={sendingOneTimePassword}
+        />
+        <BoxBody>
+          {!isVerified && sendingOneTimePassword && <Spinner />}
+          {enteringCode && <ConfirmCode
+            code={code}
+            updateCode={this.updateCode}
+          />}
+          {isVerified && <CheckIcon />}
```

Comment:
Why not just close this modal on completed verification? We are already showing success Toast.